test/plugin_loader_unit.cpp

```diff
@@ -406,6 +408,40 @@ TEST(BoostPluginLoaderUnit, LoadTestPluginsSameSymbolDifferentSections)  // NOLI
   }
 }
 
+TEST(BoostPluginLoaderUnit, TestPluginLoaderScope)  // NOLINT
```

Comment:
@Levi-Armstrong this unit test appears to work whether or not the internal cache introduced in this PR exists. According to `ldd` the unit test only links to `libboost_plugin_loader_example_plugin.so` and `libbboost_plugin_loader.so` (not the library in which the plugin was defined - `libboost_plugin_loader_example_plugin_impl.so`), so the unit test itself shouldn't be loading the plugin library as part of its runtime requirement. Any idea why this would work without holding onto the pointer to the plugin internally?